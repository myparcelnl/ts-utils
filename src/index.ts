export * from './type-guards';
export * from './types';
export * from './utils';
