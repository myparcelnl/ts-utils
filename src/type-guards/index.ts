export * from './isOfType';
export * from './isInArray';
export * from './isEnumValue';
