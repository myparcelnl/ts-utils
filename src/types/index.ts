export * from './arrayWithOneOrMore';
export * from './extractRecord';
export * from './extractRecordKeys';
export * from './makeOptional';
export * from './maybe';
export * from './newable';
export * from './oneOrMore';
export * from './promiseOr';
export * from './readonlyOr';
export * from './recursivePartial';
export * from './recursiveRequired';
export * from './replace';
export * from './requireOnly';
export * from './resolvePromise';
export * from './reverseMap';
