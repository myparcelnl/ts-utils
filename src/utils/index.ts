export * from './toArray';
export * from './asyncSome';
export * from './asyncEvery';
