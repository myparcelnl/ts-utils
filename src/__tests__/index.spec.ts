import * as EXPORTS from '../../src';
import {describe, expect, it} from 'vitest';

describe('exports', () => {
  it('should export the correct files', () => {
    expect(EXPORTS).toMatchSnapshot();
  });
});
