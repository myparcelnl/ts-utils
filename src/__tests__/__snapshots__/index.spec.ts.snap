// Vitest Snapshot v1

exports[`exports > should export the correct files 1`] = `
{
  "asyncEvery": [Function],
  "asyncSome": [Function],
  "isEnumValue": [Function],
  "isInArray": [Function],
  "isOfType": [Function],
  "toArray": [Function],
}
`;
